js_modules/dagster-ui/packages/ui-core/src/workspace/WorkspaceContext/WorkspaceContext.tsx

```diff
@@ -106,22 +106,25 @@ interface FetchLocationDataParams
   >;
 }
 
+const UNLOADED_CACHED_DATA = {};
 export const WorkspaceProvider = ({children}: {children: React.ReactNode}) => {
   const {localCacheIdPrefix} = useContext(AppContext);
   const client = useApolloClient();
   const getCachedData = useGetCachedData();
   const clearCachedData = useClearCachedData();
   const getData = useGetData();
 
-  const [locationEntryData, setLocationEntryData] = useState<
-    Record<string, WorkspaceLocationNodeFragment | PythonErrorFragment>
-  >({});
+  const [locationEntryData, setLocationEntryData] =
+    useState<Record<string, WorkspaceLocationNodeFragment | PythonErrorFragment>>(
+      UNLOADED_CACHED_DATA,
+    );
 
   const previousLocationVersionsRef = useRef<Record<string, string>>({});
 
   const {locationStatuses, loading: loadingLocationStatuses} =
     useCodeLocationStatuses(localCacheIdPrefix);
 
+  const loadingCachedData = UNLOADED_CACHED_DATA === locationEntryData;
```

Comment:
If there's no data in the indexeddb cache then on line 140 we end up assigning a new empty object reference here